src/manifolds/Circle.jl

```diff
@@ -431,6 +438,23 @@ function Random.rand(rng::AbstractRNG, ::Circle{ℝ}; vector_at=nothing, σ::Rea
         return map(_ -> σ * randn(rng), vector_at)
     end
 end
+function Random.rand(M::Circle{ℂ}; vector_at=nothing, σ::Real=1.0)
+    if vector_at === nothing
+        phi = rand() * 2 * π
+        return Complex(cos(phi), sin(phi))
```

Comment:
Better to use `sign(randn(ComplexF64))`. Why does this method not accept an `rng` argument?

Reply:
Probably because I wasn't sure if that's the right thing to do. If you think so then I will refactor at least the methods touched by this PR.

Reply:
Yes I think this is the right approach. It's the one Random itself takes.